_posts/2022-01-20-making-github-action.md

```diff
@@ -0,0 +1,131 @@
+---
+layout: blogpost
+title: Making a GitHub Action
+summary: How to create GitHub Actions to standardize data formats in repositories
+author: Daisie Huang
+date: 2022-01-20
+---
+
+At CanDIG, we work with lots of data in spreadsheet form. Many of our clinicians and data specialists expect to be able to share their data in Excel spreadsheets or similar, but these formats are less easily processed by scripts or for ingest into databases. For example, we often have to map datasets to internal data formats before they can be ingested into our system.
+
+I often wish that I could convince my more spreadsheet-oriented colleagues to export their work into a format like CSV (comma-separated values), so that I don't have to download their files to my local machine just to take a peek at the latest version. It's hard to see what changes were made in the latest upload of a binary format like xlsx: GitHub (our source code repository of choice) doesn't display xlsx files in its web interface or allow you to compare versions. 
+
+Ideally, then, our repositories would contain both xlsx files and their exported csv counterparts, so that both spreadsheet users and programmers could easily access their preferred format. However, asking users to manually update any changes to both formats before committing their files is both error-prone and possibly hard to maintain.
+
+GitHub Actions are one possible solution to maintaining this consistency. While GitHub Actions are often used for maintaining development workflows, including integration, deployment, and testing, there's no reason that we can't use Actions to automatically trigger any other required scriptable activity. For example, we can use an Action to maintain the connection between a repository's xlsx files and their csv counterparts.
+
+I created an action to do just that: https://github.com/CanDIG/xlsx2csv-action. Fundamentally, an Action is an executable, self-contained container that runs some code. The container will be executed by GitHub in an enclosing container, allowing each action's container to access the enclosing environment.The Action uses a manifest file to describe how its container will interact with the hosting container. With a GitHub Action, we can guarantee that we are executing in an environment with all of the required packages installed. Environment variables are defined to access inputs and the common workspace.
+
+So, to begin creating this action, we write a simple Python script that performs the conversion on an inputted file. This Python file, at its most basic, uses the xlsx2csv package to convert xlsx files to csvs. We can do a little more than that: the script can walk through the repository's files (temporarily stored in the action's workspace) and convert all xlsx files it finds.
```

Comment:
Can you link to the package page (maybe https://github.com/dilshod/xlsx2csv, or the pip page?)